distributed/protocol/tests/test_skorch.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+
+skorch = pytest.importorskip("skorch")
+torch = pytest.importorskip("torch")
+
+from distributed import Client
+from distributed.protocol import deserialize, serialize
+
+
+def test_serialize_deserialize_skorch_model():
+
+    client = Client(processes=True, n_workers=1)
+
+    class MyModule(torch.nn.Module):
+        def __init__(self, num_units=10):
+            super().__init__()
+            self.dense0 = torch.nn.Linear(20, num_units)
+
+        def forward(self, X, **kwargs):
+            return self.dense0(X)
+
+    net = skorch.NeuralNetClassifier(
+        MyModule,
+        max_epochs=10,
+        iterator_train__shuffle=True,
+    )
+
+    def test_serialize_skorch(net):
+        net = net.initialize()
+        return deserialize(*serialize(net))
+
+    # We test on a different worker to ensure that
+    # errors skorch serialization faces on a different process
+    # other than the client due to lack of __main__ context
+    # are actually resolved
+    # See isssue for context
+    # https://github.com/dask/dask-ml/issues/549#issuecomment-669924762
+
+    deserialized_net = list(client.run(test_serialize_skorch, net).values())[0]
+    assert isinstance(deserialized_net.module_, MyModule)
```

Comment:
Was the best test i could come up with for testing on the client. Please let me know is there is a better way to check serialization in a different process.

Reply:
I think we would benefit from a roundtrip serialization test like some of the others in this directory (without a cluster) to make sure that is working as expected. Know that doesn't show the error per-se, but it will help catch other errors in the future

In terms of testing a worker, would take a look at some of the other tests. Maybe like [this one]( https://github.com/dask/distributed/blob/13fce199ff9f956844aa926adb02a96ab4a54b6c/distributed/tests/test_worker.py#L366 )? Then adapt that to your use case. We shouldn't need to manually do the serialization ourselves, but instead rely on Dask to do that for us and merely check that things work as expected